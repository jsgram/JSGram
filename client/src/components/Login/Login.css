.header {
    flex: 1 0 auto;
}

.picture {
    height: 110px;
}

.logo {
    width: 34px;
}

.login_soft a{
    font-weight: 700;
    font-size: 18px;
}

.forgot_pass a{
    text-decoration: none;
}

.register_acc a{
    font-weight: 700;
}

