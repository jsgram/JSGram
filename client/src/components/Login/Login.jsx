import React from 'react';

export default class Login extends React.Component {

    constructor(props) {
        super(props);
        this.onEmailChange = this.onEmailChange.bind(this);
        this.onPasswordChange = this.onPasswordChange.bind(this);
    }

    onEmailChange(event) {
        this.props.setEmailText(event.target.value);
    }

    onPasswordChange(event) {
        this.props.setPasswordText(event.target.value);
    }

    render() {
        return (
            <div>
                <h3>Sign In</h3>
                <form>
                    <div>
                        <input
                            type="text"
                            name="login"
                            autoComplete="off"
                            placeholder="E-mail"
                            spellCheck={false}
                            value={this.props.email}
                            onChange={this.onEmailChange}
                        />
                    </div>
                    <div>
                        <input
                            type="password"
                            name="password"
                            placeholder="Password"
                            value={this.props.password}
                            onChange={this.onPasswordChange}
                        />
                    </div>
                    <div>
                        <button>Sign In</button>
                    </div>
                </form>
            </div>
        );
    }
}