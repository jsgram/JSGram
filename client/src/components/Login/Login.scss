.header {
    flex: 1 0 auto;
}

.picture {
    height: 110px;
}

.logo {
    width: 34px;
}

.login-soft a{
    font-weight: 700;
    font-size: 18px;
}

.forgot-pass a{
    text-decoration: none;
}

.register-acc a{
    font-weight: 700;
}

.line span {
  display: inline-block;
  height: 1px;
  width: 150px;
  background: gainsboro;
  margin: 5px;
}

