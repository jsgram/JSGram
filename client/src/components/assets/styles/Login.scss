@import './variables';

.header {
    flex: 1 0 auto;
}

.picture {
    height: 110px;
}

.logo {
    width: 34px;
}

.login_soft a{
    font-weight: 700;
    color: $color-red;
    font-size: 18px;
}

.forgot_pass a{
    text-decoration: none;
    color: $color-red;
}

.register_acc a{
    font-weight: 700;
    color: $color-red;
}