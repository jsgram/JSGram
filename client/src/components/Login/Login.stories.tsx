import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../App';
import { action } from '@storybook/addon-actions';
import Login from '../../containers/LoginContainer';

const loginState = {
    match: {
        params: {
            token: 'token',
        },
    },
};

const loginActions = {
    handleSubmit: action('handleSubmit'),
    onSubmit: action('onSubmit'),
};

storiesOf('Login', module)
    .addDecorator((story: any): any => <Provider store={store}>{story()}</Provider>)
    .addDecorator((story: any): any => <MemoryRouter>{story()}</MemoryRouter>)
    .add('login', () => <Login {...{...loginState, ...loginActions}}/>);
