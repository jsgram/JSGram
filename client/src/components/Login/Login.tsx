import React from "react";
import { Field } from "redux-form";
import { Spinner } from "reactstrap";
import { renderField } from "../commonComponents/reduxFormFields";
import { Button, Form, FormGroup } from "reactstrap";
import "../styles/Login.scss";
import logo from "../assets/logo.png";

export default class Login extends React.Component<any> {
  render() {
    const { handleSubmit, onSubmit, submitting } = this.props;
    return (
      <div className="container-fluid header">
        <div className="row justify-content-center align-items-center">
          <div className="col-sm-8 col-md-6 col-xl-5">
            <Form className="mt-4 bg-white" onSubmit={handleSubmit(onSubmit)}>
              <div className="border">
                <FormGroup className="col-lg-10 offset-lg-1 text-center">
                  <img className="picture" src={logo} alt="logo" />
                  <Field
                    className="form-control form-control-lg"
                    type="text"
                    name="email"
                    autoComplete="off"
                    placeholder="E-mail"
                    spellCheck={false}
                    component={renderField}
                  />
                  <Field
                    className="form-control form-control-lg mt-3"
                    type="password"
                    name="password"
                    placeholder="Password"
                    component={renderField}
                  />
                  <Button
                    className="mt-3"
                    color="danger"
                    disabled={submitting}
                    size="lg"
                    block
                  >
                    Log In
                  </Button>
                  {submitting && <Spinner color="dark" />}
                </FormGroup>
                <div className="d-flex justify-content-around line">
                  <div>
                    <span />
                    OR
                    <span />
                  </div>
                </div>
                <div className="text-center mt-2">
                  <p>
                    <img
                      className="logo pb-1"
                      src="https://www.armstrongsgroup.com/wp-content/uploads/2017/03/facebook-logo-black-and-white-png.png"
                      alt="Facebook logo"
                    />
                    <a href="#" className="text-danger login-soft">
                      Log in with Facebook
                    </a>
                  </p>
                  <p>
                    <a href="#" className="text-danger forgot-pass">
                      Forgot password?
                    </a>
                  </p>
                </div>
              </div>
            </Form>
            <Form className="bg-white mt-3">
              <div className="border">
                <FormGroup className="text-center register-acc mt-2">
                  <p className="pt-2">
                    Still don't have an account?
                    <a href="#" className="pl-1 text-danger">
                      Register
                    </a>
                  </p>
                </FormGroup>
              </div>
            </Form>
          </div>
        </div>
      </div>
    );
  }
}
