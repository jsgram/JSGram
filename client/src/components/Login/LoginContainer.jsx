import React from 'react';
import {connect} from 'react-redux';
import Login from "./Login";
import {setEmailText, setPasswordText} from "../../store/login/actions";

class LoginContainer extends React.Component {
    render() {
        return <Login email={this.props.email} password={this.props.password} setEmailText={this.props.setEmailText}
                      setPasswordText={this.props.setPasswordText}/>;
    }
}

const mapStateToProps = (state) => {
    return {
        email: state.login.email,
        password: state.login.password
    };
};

const mapDispatchToProps = {
    setEmailText,
    setPasswordText
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);