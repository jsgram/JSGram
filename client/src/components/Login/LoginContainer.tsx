import React from 'react';
import Login from './Login';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { loginUser } from '../../store/login/actions';
import validate from '../../utils/validation';
import { FormProps } from 'reactstrap';
import { IUser } from '../../store/commonInterfaces/commonInterfaces';
import {setToken} from '../../store/login/setToken.helper';

export class LoginContainer extends React.Component<any> {
    constructor(props: any) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
    }

    public onSubmit(user: IUser): void {
        return this.props.loginUser(user);
    }

    // TODO Refactor after authorization
    public componentDidMount(): void {
        const {token}: { token: string } = this.props.match.params;
        if (token) {
            setToken(token);
        }
    }

    public render(): JSX.Element {
        const {handleSubmit, submitting}: FormProps = this.props;
        return (
            <Login
                handleSubmit={handleSubmit}
                onSubmit={this.onSubmit}
                submitting={submitting}
            />
        );
    }
}

export default connect(
    null,
    {loginUser},
)(
    reduxForm({
        form: 'loginForm',
        validate,
    })(LoginContainer),
);
