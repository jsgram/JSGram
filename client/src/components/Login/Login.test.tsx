import Login from './index';
import * as reactstrap from 'reactstrap';

import { shallow } from 'enzyme';
import React from 'react';

describe('Login component:', () => {
    let renderer;

    beforeEach(() => {
        const mockOnSubmit = jest.fn(() => 'somevalue');
        const mockHandleSubmit = jest.fn(() => 'somevalue');

        reactstrap.Spinner = jest.fn(() => (<div></div>));
        reactstrap.Button = jest.fn(() => (<div></div>));
        reactstrap.Form = jest.fn(() => (<div></div>));
        reactstrap.FormGroup = jest.fn(() => (<div></div>));

        renderer = shallow(<Login handleSubmit={mockHandleSubmit} onSubmit={mockOnSubmit} />);
    });

    test.skip('render - success', () => {
        expect(renderer.html()).toHaveLength(168);
    });
});
