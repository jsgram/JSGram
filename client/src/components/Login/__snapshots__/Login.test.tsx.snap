// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component: render - success 1`] = `ShallowWrapper {}`;
